src/modules/did/did.service.ts

```diff
@@ -99,49 +109,124 @@ export class DIDService {
+    return documentFromLogs(id, [logs]);
+  }
+
+  /**
+   * Reads all logs for a given DID that haven't been cached yet and merged them
+   * @param documentEntity
+   */
+  private async updateLogs(
+    id: string,
+    logs: IDIDLogData,
+  ): Promise<IDIDLogData> {
+    if (logs?.topBlock) {
+      const readFromBlock = logs.topBlock.add(1); // Want to read from 1 more than previously last read to
+      const newLogs = await this.resolver.readFromBlock(id, readFromBlock);
+      const mergedLogs = mergeLogs([logs, newLogs]);
+      mergedLogs.topBlock = newLogs.topBlock;
+      return mergedLogs;
+    }
+    const newLogs = await this.getAllLogs(id);
+    const mergedLogs = mergeLogs([logs, newLogs]);
+    mergedLogs.topBlock = newLogs.topBlock;
+    return mergedLogs;
+  }
+
+  /**
+   * Gets all logs for a given DID
+   * @param documentEntity
+   */
+  private async getAllLogs(id: string): Promise<IDIDLogData> {
+    const genesisBlockNumber = 0;
+    const readFromBlock = utils.bigNumberify(genesisBlockNumber);
+    return this.resolver.readFromBlock(id, readFromBlock);
+  }
+
+  private updateServices(
+    services: IServiceEndpoint[],
+    cachedServices: IClaim[] = [],
+  ): Promise<IClaim[]> {
+    return Promise.all(
+      services.map(async ({ serviceEndpoint, ...rest }) => {
+        const cachedService = cachedServices.find(
+          claim => claim.serviceEndpoint === serviceEndpoint,
+        );
+        if (cachedService) return cachedService;
+
+        const token = await this.ipfsStore.get(serviceEndpoint);
+
+        const { claimData, ...claimRest } = jwt.decode(token) as {
+          claimData: Record<string, string>;
+        };
+        return {
+          serviceEndpoint,
+          ...rest,
+          ...claimData,
+          ...claimRest,
+        };
+      }),
+    );
+  }
+
   /**
    * Retrieves a DID Document for a given DID
    * @param {DID} did DID whose document should be retrieved
    * @param {boolean} enhanceWithClaims
    * @returns {IDIDDocument} Resolved DID Document. Null if no Document is not cached.
    */
-  public async getById(
-    did: DID,
-    enhanceWithClaims = false,
-  ): Promise<IDIDDocument> {
-    let cachedDIDDocument = await this.didRepository.findOne(did.id);
-    if (!cachedDIDDocument) {
-      this.logger.info(
-        `Requested document for did: ${did.id} not cached. Queuing cache request.`,
-      );
-      // awaiting refresh so that subsequent calls don't duplicate cached DID Document
-      await this.refreshCachedDocument(did);
-      cachedDIDDocument = await this.didRepository.findOne(did.id);
-    }
-    const didEntity = new DIDDocumentEntity(did, cachedDIDDocument);
-    const resolvedDocument = await didEntity.getResolvedDIDDocument();
-    if (enhanceWithClaims) {
-      this.enhanceWithClaims(didEntity, resolvedDocument);
-    }
-    return resolvedDocument;
+  public async getById(did: string): Promise<IDIDDocument> {
```

Comment:
A bit of a personal preference, but I would put public methods before private in the file

Reply:
👍